RFS/src/test/java/com/rfs/common/LuceneDocumentsReaderTest.java

```diff
@@ -0,0 +1,95 @@
+package com.rfs.common;
+
+import static org.mockito.Mockito.*;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.util.BytesRef;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+class TestLuceneDocumentsReader extends LuceneDocumentsReader {
+    // Helper method to correctly encode the Document IDs for test
+    public static byte[] encodeUtf8Id(String id) {
+        byte[] idBytes = id.getBytes(StandardCharsets.UTF_8);
+        byte[] encoded = new byte[idBytes.length + 1];
+        encoded[0] = (byte) Uid.UTF8;
+        System.arraycopy(idBytes, 0, encoded, 1, idBytes.length);
+        return encoded;
+    }
+
+    // Override to return a mocked IndexReader
+    @Override
+    protected IndexReader openIndexReader(Path indexDirectoryPath) throws IOException {
+        // Set up our test docs
+        Document doc1 = new Document();
+        doc1.add(new StringField("_id", new BytesRef(encodeUtf8Id("id1")), Field.Store.YES));
+        doc1.add(new StoredField("_source", new BytesRef("source1")));
+
+        Document doc2 = new Document();
+        doc2.add(new StringField("_id", new BytesRef(encodeUtf8Id("id2")), Field.Store.YES));
+        doc2.add(new StoredField("_source", new BytesRef("source2")));
+
+        Document doc3 = new Document();
+        doc3.add(new StringField("_id", new BytesRef(encodeUtf8Id("id3")), Field.Store.YES));
+        doc3.add(new StoredField("_source", new BytesRef("source3")));
+
+        Document doc4 = new Document(); // Doc w/ no fields
+        
+        Document doc5 = new Document(); // Doc w/ missing _source
+        doc5.add(new StringField("_id", new BytesRef(encodeUtf8Id("id5")), Field.Store.YES));
+
+        // Set up our mock reader
+        IndexReader mockReader = mock(IndexReader.class);
```

Comment:
Let's sync-up on mockito.  I wonder if this could have been clearer and tighter without mockito.